Hide Children/ChildHider.js

```diff
@@ -161,54 +165,91 @@ tau.mashups
                 $.ajax({
-                    url: configurator.getApplicationPath() + '/storage/v1/childHider/U' +
-                        loggedUser.id,
+                    url: this._getStorageUrl(),
                     method: 'GET'
-                }).then(_.bind(function(response) {
-                    if (_.has(response.userData, this.boardId)) {
-                        if ((response.userData[this.boardId] == "0") != (!this.state)) {
-                            this.toggle();
+                }).then(function(response) {
+                    var userData = response.userData;
+                    if (_.has(userData, this.boardId)) {
+                        // '1' - hideChildren === true
+                        // '0' or undefined - hideChildren === false
+                        var shouldHideChildren = userData[this.boardId] === '1';
+                        if (shouldHideChildren !== this.hideChildren) {
+                            this.hideChildren = shouldHideChildren;
+                            this.updateCardsVisibility();
+                            this.refreshBoardSize();
                         }
                     }
-                }, this));
+                }.bind(this));
             },
 
             saveState: function() {
-                var data = {
-                    "userData": {}
-                };
-                data['userData'][this.boardId.toString()] = this.state ? "1" : "0";
+                var data = {userData: {}};
+                data.userData[this.boardId.toString()] = this.hideChildren ? '1' : '0';
                 $.ajax({
-                    url: configurator.getApplicationPath() + '/storage/v1/childHider/U' +
-                        loggedUser.id,
+                    url: this._getStorageUrl(),
                     method: 'POST',
                     data: JSON.stringify(data),
                     contentType: 'application/json; charset=utf8'
                 });
             },
 
             toggle: function() {
-                this.state = !this.state;
-                _.each(this.represented, _.bind(function(id) {
-                    $('div[role=card][data-entity-id=' + id + ']')[this.state ? 'hide' :
-                        'show']();
-                }, this));
-                this.$btn.html(this.state ? 'Show Children (' + this.represented.length + ')' :
-                    'Hide Children');
+                this.hideChildren = !this.hideChildren;
                 this.saveState();
+                this.updateCardsVisibility();
+                this.refreshBoardSize();
+            },
 
-                refreshBoardSize();
+            updateCardsVisibility: function() {
+                var $cards = $('.i-role-card');
+                var $cardsToHide = $();
+                var cardsCount = 0;
 
+                _.each(this.childrenIds, function(id) {
+                    var $cardToHide = $cards.filter('[data-entity-id=' + id + ']');
```

Comment:
Filter with access to data attribute is applied to elements collection. Filter by integer Id's array seem to be faster

Try someting like:
```js
var $cardToHide = _.chain($('.i-role-card')).reduce(function(aggregate, $card) {
var entityId = $card.data('entity-id');
if(this.childrenIds.indexOf(entityId) >=0)
{
     aggregate[entityId] = $card;
}
return aggregate;
}, {}).mapObject(function(card, id) {
     return card;
}).value();
```